packages/firebase_crashlytics/lib/src/firebase_crashlytics.dart

```diff
@@ -0,0 +1,190 @@
+    if (!enableInDevMode) {
+      assert(_inDebugMode = true);
+    }
+    return _inDebugMode;
+  }
+
+  static const MethodChannel channel =
+      MethodChannel('plugins.flutter.io/firebase_crashlytics');
+
+  Future<void> onError(FlutterErrorDetails details) async {
+    print('Error caught by Crashlytics plugin:');
+    if (isInDebugMode && !enableInDevMode) {
+      print(Trace.format(details.stack).trimRight().split('\n'));
+    } else {
+      // Send logs
+      await sendLogs();
+
+      // Send keys
+      await sendKeys();
+
+      // Report error
+      final List<String> stackTraceLines =
+          Trace.format(details.stack).trimRight().split('\n');
+      final dynamic result =
+          // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+          // https://github.com/flutter/flutter/issues/26431
+          // ignore: strong_mode_implicit_dynamic_method
+          await channel.invokeMethod('Crashlytics#onError', <String, dynamic>{
+        'exception': details.exceptionAsString(),
+        'stackTrace': details.stack.toString(),
+        'stackTraceLines': stackTraceLines,
+        'code': stackTraceLines[0].hashCode
+      });
+      print(result);
+    }
+  }
+
+  void crash() {
+    throw StateError('Error thrown by Crashlytics plugin');
+  }
+
+  /// Reports the global value for debug mode.
+  /// TODO(kroikie): Clarify what this means in context of both Android and iOS.
+  Future<bool> isDebuggable() async {
+    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+    // https://github.com/flutter/flutter/issues/26431
+    // ignore: strong_mode_implicit_dynamic_method
+    final bool result = await channel.invokeMethod('Crashlytics#isDebuggable');
+    return result;
+  }
+
+  Future<String> getVersion() async {
+    // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+    // https://github.com/flutter/flutter/issues/26431
+    // ignore: strong_mode_implicit_dynamic_method
+    final String result = await channel.invokeMethod('Crashlytics#getVersion');
+    return result;
+  }
+
+  /// Add text logging that will be sent with your next report. `msg` will be
+  /// printed to the console when in debug mode.
+  void log(String msg) {
+    _logSize += Uint8List.fromList(msg.codeUnits).length;
+    logs.add(msg);
+    // Remove oldest log till logSize is no more than 64K.
+    while (_logSize > 65536) {
+      final String first = logs.removeFirst();
+      _logSize -= Uint8List.fromList(first.codeUnits).length;
+    }
+  }
+
+  void setKey(String key, dynamic value) {
+    // Check that only 64 keys are set.
+    if (keys.containsKey(key) || keys.length <= 64) {
+      keys[key] = value;
+    }
+  }
+
+  void setBool(String key, bool value) {
+    setKey(key, value);
+  }
+
+  void setDouble(String key, double value) {
+    setKey(key, value);
+  }
+
+  void setInt(String key, int value) {
```

Comment:
`setInt`, `setBool` & `setDouble` are purely convenience methods on native SDKs (iOS/Android) and internally the SDK's convert these to a string value. 

Perhaps it would be better to simplify this and have a singular method in Dart that accepts a `bool`, `double`, `int` or `String` value and this method converts to a String in dart and calls only one native method (`setString`)

Reply:
@Salakar thanks for the feedback. We are trying to keep the plugins interface as close to the native ones as possible. So we will stick with the separate methods for now.